ui/appui-react/src/appui-react/widgets/Widget.tsx

```diff
@@ -52,4 +52,8 @@ export interface Widget {
   readonly label?: string | ConditionalStringValue;
   readonly priority?: number;
   readonly tooltip?: string | ConditionalStringValue;
+  /** Describes to which container the item is added.
+   * @alpha TODO: rename due to CanFloatWidgetOptions["containerId"]
```

Comment:
Suggestions welcome.

Reply:
what exactly is a container in this sense?

Reply:
Similar to `toolbarId` it would map to a section of a stage panel (replaces `StagePanelLocation` and `StagePanelSection` args)

Reply:
I think its dangerous to expect this field which is typed string to be of the form `JSON.stringify({ location, section })`

I would think it'd be easier to manage by just having a `container` field that maps to `{ location: SPL; section: SPS; }`

A part of me thinks adding a layer with the widget and this would help for future support of layouts that have different location/section concepts (you might not want to have those on the actual widget) but there's already allowedPanels so probably not a big deal

Reply:
It would be easier to manage, but not as extensible. I.e. suppose we'd want to provide a separate layout that supports previously removed "topMost" location or "middle" section. On the other hand it would probably make sense to have this strongly typed instead (and i.e. allow any object for customization?).

Reply:
If we do that, then all providers needs to be updated to support all layouts, and we cannot allow new layouts to be defined as we still stick to a specific sematic. I dont think that `JSON.strignify` is a solution however, but we could provide wellknowns enums to be used, on top of any arbitrary strings.

``` ts
enum AppUIWidgetTargets {
  TopStart: "appuidefaultlayout-top-start",
  TopEnd: "appuidefaultlayout-top-end",
  ...
}
```

Then we can use these in the `containerId` the well known ones, and if you want something else, you are free to do so.

``` ts
containerId: AppUIWidgetTargets.TopEnd;
```

Then the layout can ask for widgets that are directed to itself.

Reply:
I tend to agree with @jason-crow on the union types, however, since we want to support "custom" layouts we might not want to add SpatialWidget and PanelsWidget on the same type and instead consider a separate `SpatialUiItemsProvider` type... As discussed, it would make sense to extract layout-agnostic Widget properties. The biggest challenge is backward compatibility.

Reply:
Introduced `PanelWidget` with specific properties to replace the `containerId`.